temporalio/workflow.py

```diff
@@ -415,6 +416,56 @@ def decorator(
     return decorator(fn.__name__, description, fn, bypass_async_check=True)
 
 
+@dataclass(frozen=True)
+class DynamicWorkflowConfig:
+    """Returned by functions using the :py:func:`dynamic_config` decorator, see it for more."""
+
+    failure_exception_types: Sequence[Type[BaseException]] = dataclasses.field(
```

Comment:
Kinda too bad this is dynamic only, this is a nice feature regardless heh (as would be a callback where they can choose programmatically whether an exception is task fail or workflow fail)

Reply:
Yeah, I mean we can shift it to be like that if people ask I suppose - but since it only applies on first WFT for now it's not much different from the defn version